File: maths/Codes/Fibonacci_Sequence_Generator.py
def into_integer(x, msg='Cannot be converted into a integer'):
    try:
        x = int(x)
    except ValueError:
        print(msg)
    else:
        return x


terms = get_integer('No of terms: ')
prev_first = 0
prev_second = 1

print(prev_first)
print(prev_second)

while 2 < terms:
    new = prev_second + prev_first
    prev_first = prev_second
    prev_second = new
    print(new)
    terms -= 1

input("Press enter to exit...")

File: maths/codes/download/Fibonacci Sequence Generator.cpp
#include <iostream>

int get_integer(string msg = "Number: ") {
    int i;
    cout << msg;
    cin >> i;
    return i;
}

using namespace std;

int main() {
    int terms = get_integer("No of terms: ");
    int prev_first = 0, prev_second = 1;
    cout << prev_first << endl;
    cout << prev_second << endl;

    while (2 < terms) {
        int n = prev_second + prev_first;
        prev_first = prev_second;
        prev_second = n;
        cout << n << endl;
        terms -= 1;
    }
}